### ISSUE TITLE
Allow to easily integrate in an existing Elixir app

### ISSUE THREAD

**Issue by jfrolich** [papercups-io/papercups#210]
It would be great if most of this can be moved to a `mix` app where it can be easily integrated inside of another app. This repo can just use that dependency.

I also saw that the table names are not namespaced. Actually namespacing these would also help to have this coexist.

**Comment by reichert621** [comment]
oh man this would be really cool... i think we probably won't have time to do this for a while since there's a bunch of other things on our current roadmap, but i hadn't though about this until now...

just to make sure i understand correctly, what you'd like to see is a `papercups` dependency that you can install, and then invoke methods like `Papercups.Messages.list(...)` and stuff like that?

**Comment by jfrolich** [comment]
Yeah basically have the whole app inside of an existing app. This is possible with Elixir as it's quite composable. (Basically you just need to add some lines to routes? And maybe some global config?).
If we look at adopting papercups I'll probably try to make a PR to enable this.

**Comment by Grantimus9** [comment]
@jfrolich I think including the js is the least difficult part of such a transformation, but to include the js widget, you could do what phoenix does: include the js in the hex package, and reference it from the app's package.json like so: 
 
```javascript
  "dependencies": {
    "phoenix": "file:../deps/phoenix", 
    "papercups_phx_widget": "file:../deps/papercups"
}
```
And then later in js files import things like so: 
```javascript
import {Socket} from "phoenix"
import Widget from "papercups_phx_widget" 
```

Maybe something that would get one most of the way is a hex package that always includes the latest widget js bundle and enables existing phx apps to include the dependency, provide some config for their current papercups.io account, and then that would enable an existing phx app to have all traffic from the widget go through its own server first, proxying everything to the papercups.io server. This enables webapps to reduce the number of third party http calls from their page.
